Makefile.am

```diff
@@ -226,11 +244,18 @@ samples_camera_sample_mavlink_client_SOURCES = \
 	src/util.c \
 	src/util.h
 
+samples_camera_sample_mavlink_client_LDADD = $(GLIB_LIBS)
+if !DISABLE_AVAHI
+samples_camera_sample_mavlink_client_LDADD += $(AVAHI_LIBS)
```

Comment:
No. If I add that line on first !DISABLE_AVAHI block it will fail if mavlink is disabled.
this libs should be added only if AVAHI and MAVLink are enabled.